x11-misc/libinput-gestures/libinput-gestures-2.29.ebuild

```diff
@@ -0,0 +1,32 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{3_4,3_5,3_6} )
+
+inherit python-single-r1
+
+DESCRIPTION="Actions gestures on your touchpad using libinput"
+HOMEPAGE="https://github.com/bulletmark/${PN}"
+SRC_URI="https://github.com/bulletmark/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="gtk kde"
+
+RDEPEND="${PYTHON_DEPS}
+	>=dev-libs/libinput-1.8.0
+	x11-misc/xdotool
+	x11-misc/wmctrl"
+DEPEND=">=dev-libs/libinput-1.8.0
+	dev-util/desktop-file-utils
+	gtk? ( x11-libs/gtk+:3 )
+	kde? ( kde-plasma/kde-cli-tools:5 )"
```

Comment:
What is kde-cli-tools doing in DEPEND (but not RDEPEND)?

Reply:
There is a script libinput-gestures-setup in this package. Upstream uses it as another way to launch libinput-gestures. I added kde-plasma/kde-cli-tools to ebuild to support this script too.

     if echo $XDG_CURRENT_DESKTOP | grep -q KDE; then
        if hash kioclient5 &>/dev/null; then
            kioclient5 exec "$APPDIR/$app.desktop" >/dev/null
            return $?
        elif hash kioclient &>/dev/null; then
            kioclient exec "$APPDIR/$app.desktop" >/dev/null
            return $?
        fi
    fi

kioclient5 is not needed during libinput-gestures runtime, and thus kde-plasma/kde-cli-tools is not needed too.

Reply:
That script is run depending on the value of `$XDG_CURRENT_DESKTOP` at build time?

Reply:
Yes, it depends on a value of $XDG_CURRENT_DESKTOP. For KDE it runs kioclient, for GTS - gtk-launch. But all this not necessarily happens at build time, since ebuild does not use this script.

It depends from the point of view.
If you treat libinput-gestures-setup as a part of build system (which this package doesn't have from my point o view ), then x11-libs/gtk+ and kde-plasma/kde-cli-tools should be in DEPEND.
If you treat libinput-gestures-setup as a part of a package (like init script for example), then these should go to RDEPEND.

>  ~ $ libinput-gestures-setup 
> Usage:
> As root: sudo libinput-gestures-setup install|uninstall
> As user: libinput-gestures-setup start|stop|restart|autostart|autostop|status
> 
> (-d option sets DESTDIR for install/uninstall)

Reply:
By your description, yes, those should be in RDEPEND, and not in DEPEND. However, Portage does not support optional runtime dependencies. I would therefore recommend to drop them as deps and rather add a has_version check and elog message in pkg_postinst.

Reply:
@SpiderX any movement here?